### DISCUSSION TITLE
Files greater than 2GB as Embedded Resource

### DISCUSSION THREAD

**Discussion by ctrl-vee** [0xd4d/dnlib#475] [Q&A]
Hello! I was wondering how you use the DataReaderFactory to create a new Embedded Resource for a file that is larger than 2GB? (An executable or .DLL in this case) 

With smaller files I’m able to use File.ReadAllBytes, how can I set it up for this use case ?

**Comment by wtfsck** [comment]
.NET arrays are limited to 2GB in size so that's not possible.
